crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -632,21 +642,21 @@ impl InspectorStackRefMut<'_> {
             };
 
             // set depth to 1 to make sure traces are collected correctly
-            evm.context.evm.inner.journaled_state.depth = 1;
+            evm.journaled_state.depth = 1;
 
-            let res = evm.transact();
+            let res = evm.transact(env.tx.clone());
 
             // need to reset the env in case it was modified via cheatcodes during execution
-            ecx.env = evm.context.evm.inner.env;
+            ecx.cfg = cached_env.evm_env.cfg_env;
+            ecx.block = cached_env.evm_env.block_env;
+            ecx.tx = cached_env.tx;
+
```

Comment:
This is likely incorrect and should be replaced by 

```rust
            let res = evm.transact(evm.tx.clone());

            // apply the changes to the env
            *env.cfg = evm.cfg.clone();
            *env.block = evm.block.clone();
            *env.tx = evm.tx.clone();
 ```

Reply:
Note: this currently results in a underflow of a gas refund, investigating

Reply:
Resolved in https://github.com/foundry-rs/foundry/pull/10183/commits/30c93ea51fed565abd77f7a25fe5c4ce545a6d31

Restoring behavior to https://github.com/foundry-rs/foundry/blob/a34f4c989b94f572497631ff5c85909d674c23a6/crates/evm/evm/src/inspectors/stack.rs#L640-L649 equivalent